source/Octopus.Tentacle.Tests.Integration/Support/ClientAndTentacle.cs

```diff
@@ -42,15 +46,24 @@ public LegacyTentacleClientBuilder LegacyTentacleClientBuilder(AsyncHalibutFeatu
             TemporaryDirectory = temporaryDirectory;
             RpcRetrySettings = rpcRetrySettings;
             this.ServiceEndPoint = serviceEndPoint;
+            this.logger = logger.ForContext<ClientAndTentacle>();
         }
 
         public async ValueTask DisposeAsync()
         {
-            Server.Dispose();
-            PortForwarder?.Dispose();
-            await RunningTentacle.DisposeAsync();
+            logger.Information("Starting DisposeAsync");
+
+            logger.Information("Starting RunningTentacle.DisposeAsync and Server.Dispose and PortForwarder.Dispose");
+            var portForwarderTask = Task.Run(() => PortForwarder?.Dispose());
+            var runningTentacleTask = RunningTentacle.DisposeAsync();
+            var serverTask = Server.DisposeAsync();
+            await Task.WhenAll(runningTentacleTask.AsTask(), serverTask.AsTask(), portForwarderTask);
```

Comment:
Just a thought, but in the above output, it showed that `Starting Server.Dispose` was the one that took 40 seconds.

But now that we're doing it in parallel, we won't know in future which is the item that's taking the longest.

Is it worth starting a new task and logging the start/end for each of these?

It might be overkill, but worth mentioning at least 😁

Reply:
Whatever we decide here, we can do elsewhere as well.

Reply:
Through a combo of places that log we do end up with enough info to see where each task is up to and if it has finished

```
00:00:24.4383801 [ClientAndTentacle] Starting DisposeAsync
00:00:24.4383867 [ClientAndTentacle] Starting RunningTentacle.DisposeAsync and Server.Dispose and PortForwarder.Dispose

00:00:24.4387972 [PortForwarder] Start Dispose Pumps
00:00:24.4388022 [PortForwarder] Fisinshed Dispose Pumps

00:00:24.4390581 [RunningTentacle] Starting DisposeAsync
00:00:24.4390639 [RunningTentacle] Starting Stop

00:00:24.4468457 [Server] Starting ServerHalibutRuntime.DisposeAsync

00:00:24.4518393 [RunningTentacle] Starting deleteInstanceFunction

00:00:24.4642509 [Server] Finished ServerHalibutRuntime.DisposeAsync

00:00:25.1240382 [RunningTentacle] [Tentacle] Deleted instance: TentacleIT-29fec592368d4c27a5d14ad2f778d0c4
00:00:25.1661693 [RunningTentacle] Starting temporaryDirectory.Dispose
00:00:25.1682434 [RunningTentacle] Finished DisposeAsync

00:00:25.1682548 [ClientAndTentacle] Starting TentacleClient.Dispose
00:00:25.1682587 [ClientAndTentacle] Starting TemporaryDirectory.Dispose
00:00:25.1685010 [ClientAndTentacle] Finished DisposeAsync

00:00:25.1685894 [ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled] Staring Test Tearing Down
00:00:25.1685935 [ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled] Cancelling CancellationTokenSource
00:00:25.1685976 [ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled] Disposing CancellationTokenSource
00:00:25.1686013 [ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled] Finished Test Tearing Down
```